httpx/_types.py

```diff
@@ -93,7 +93,7 @@
 
 RequestData = Mapping[str, Any]
 
-FileContent = Union[IO[bytes], bytes, str]
+FileContent = Union[IO[bytes], bytes, str, AsyncIterable[bytes], AsyncIterable[str]]
```

Comment:
> Is there an async equivalent to typing.IO?

I don't think so. Since all the asyncio files I know are implemented in 3rd party libraries, there is no need for `typing.IO` asynchronous alternative in stdlib, I guess.

> Note that only support byte streams, not text streams.

But it kinda works in that case. I'm using `yield to_bytes(achunk)` in `arender_data` to convert `str` chunk to bytes. I've updated the test to work with the file in text mode also. Not sure if this is correct aproach =/.

Reply:
Ah yeah, I think the failure case is this?...

* Read file size, used for Content-Length header.
* Open a file on a system using a non utf-8 encoding.
* Write to bytes using utf-8.
* Oops, uploaded data stream didn't match the Content-Length.

Reply:
Ok, I've updated the code and added a runtime checks for bytes in `arender_data` and updated tests accordingly.